src/patterns/flyweight/avatar-example/example/utils.ts

```diff
@@ -1,10 +1,14 @@
 import {ALPHABETIC_COLORS_MAP, AlphabetLetter, IAvatar, IBaseContact} from "./";
 
-export const generateKey = (sharedState: IBaseContact): string => {
-  return sharedState.name.charAt(0);
+export const generateKey = (sharedState: IBaseContact): AlphabetLetter => {
+  return sharedState.name.charAt(0) as AlphabetLetter;
 };
 
-export const generateIntrinsicState = (sharedState: IBaseContact): IAvatar => ({
-  initialLetter: generateKey(sharedState),
-  color: ALPHABETIC_COLORS_MAP.get(generateKey(sharedState) as AlphabetLetter) as string,
-});
+export const generateIntrinsicState = (sharedState: IBaseContact): IAvatar => {
+  const key = generateKey(sharedState);
+
+  return ({
+    initialLetter: key,
+    color: ALPHABETIC_COLORS_MAP.get(key) ?? ALPHABETIC_COLORS_MAP.get('default')!,
```

Comment:
@MaslovMykhailo 
Even If I try `ALPHABETIC_COLORS_MAP.has(key)` firstly the method `Map.get(): T : undefined` still treats the possibly undefined result.